modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -78,21 +77,19 @@ def parse_dacl_or_sacl(acl)
 
       # To decode the ObjectType we need to do another query to CN=Configuration,DC=daforest,DC=com
       # and look at either schemaIDGUID or rightsGUID fields to see if they match this value.
-      next unless ace_body[:flags] && ace_body[:flags][:ace_object_type_present] == 1
+      if (object_type = ace_body[:object_type]) && !(object_type == CERTIFICATE_ENROLLMENT_EXTENDED_RIGHT || object_type == CERTIFICATE_AUTOENROLLMENT_EXTENDED_RIGHT)
+        # If an object type was specified, only process the rest if it is one of these two
+        next
+      end
 
-      object_type = ace_body[:object_type]
+      next if (ace_body.access_mask.protocol & CONTROL_ACCESS) == 0
```

Comment:
Might want to add a comment here explaining what this is for, as I myself have forgotten why this was added in.

Reply:
Will update this in another commit.

Reply:
Added in comments in https://github.com/rapid7/metasploit-framework/pull/17968/commits/0441c1f18760ddc876736ddd138d1bee9a91f35c